docs/search_syntax.md

```diff
@@ -0,0 +1,281 @@
+# Search Syntax Documentation
+## Introduction
+
+This document provides an overview of the search syntax that users can utilize to filter metrics and runs.
+
```

Comment:
We should start by introducing that the syntax is essentially Python "so it should feel pretty intuitive for most users", and then explain that we'll dive into more details about supported operations below.